src/main/java/library/app/controller/BookController.java

```diff
@@ -1,20 +1,31 @@
 package library.app.controller;
 
+import library.app.dao.BookDao;
+import library.app.entity.Book;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
 @Controller
 @RequestMapping("/book")
 public class BookController {
 
+    @Autowired
+    private BookDao bookDao;
+
     @GetMapping
-    public String getAllBooks() {
+    public String getAllBooks(ModelMap model) {
+        model.put("books", bookDao.listBooks());
         return "allBooks";
     }
 
     @GetMapping("/info")
-    public String bookInfo() {
+    public String bookInfo(ModelMap model, @RequestParam Long id) {
```

Comment:
in next homework try to use `PathVariable`?

Reply:
OK. But Bogdan told today, we'd better use RequestParam for id

Reply:
copy from slack:
book info - use @ PathVariable

Reply:
Sorry, mixed up :)